### PR TITLE
v1.24 blog post: Volume populators beta

### PR DESCRIPTION
**Pull request by bswartz**

### CODE REVIEW

**Comment by katcosgrove** [comment]
Hey there! I'm from the Comms team, and I wanted to check on the status of this blog. Reviews for feature blogs are supposed to start next week. Anything we can help you with?

**Comment by bswartz** [comment]
> Hey there! I'm from the Comms team, and I wanted to check on the status of this blog. Reviews for feature blogs are supposed to start next week. Anything we can help you with?

Hey, thanks for checking in! As this blog is about a feature moving from alpha to beta, the blog entry will partly reprise the original alpha blog, with updates related to the changes in beta, and also mention the additional features included with beta (mostly addition of metrics).

My goal with the second blog entry is to remind people that this feature is still here, and now switched on by default, and to encourage more usage. There's only a handful of real-world usage examples I can point to, but I'll try to spark people's imagination with our (SIG storage's) future plans for this feature.

**Comment by sftim** [comment]
I've suggested (elsewhere) that we try to combine the storage-related changes for v1.24 into one, or maybe two, articles. This feels like a good candidate for that if it would recap existing announcements.

PS. If the previous article is now carrying a “this article is more than a year old and might be out of date” banner, or you're worried that it soon will be, there are things we can look at to make the text a bit more evergreen and then suppress the banner for that specific article.

**Comment by bswartz** [comment]
Fixed a spelling error, and a URL link

**Comment by sftim** [comment]
I just noticed that this PR targets _dev-1.24_

For a blog article, we actually want the PR to target _main_ - could you switch the base branch over?

File: content/en/blog/_posts/2022-04-12-volume-populators-beta.md [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+associated populator controller in the cluster, and any user who can create instances of 
+the CR can create pre-populated volumes by taking advantage of the populator.
+
+Multiple populators can be installed side by side for different purposes. The SIG storage
+community is already seeing some implementations in public, and more prototypes should
+appear soon.
+
+Cluster administrations are **strongly encouraged** to install the
+volume-data-source-validator controller and associated `VolumePopulator` CRD, so that users
+can get feedback about invalid PVC data sources.
```
[the remark this answers is not in the document]
Comment (bswartz) [reply content/en/blog/_posts/2022-04-12-volume-populators-beta.md:28]: While it's possible to run a populator without this controller, we expect every populator to create a `VolumePopulator` CR as part of its installation, and if the CRD is not installed, that step will fail. Therefore we really don't want to see populators getting installed on clusters that don't have the volume-data-source-validator.

It's true that if you don't have any populators, then the only downside of not installing the controller is that users won't get feedback if they mistakenly try to use a populator that isn't there. Maybe we can rephrase that we strongly recommend it if you also install any populators?

**Comment by bswartz** [comment]
@sftim Updated based on your comments

File: content/en/blog/_posts/2022-04-12-volume-populators-beta.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (sftim) [reply content/en/blog/_posts/2022-04-12-volume-populators-beta.md]: The revised wording is better.

File: content/en/blog/_posts/2022-04-12-volume-populators-beta.md [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+layout: blog
+title: "Kubernetes 1.24: Volume Populators Graduate to Beta"
+date: 2022-04-12
+slug: volume-populators-beta
+---
+
+**Authors:**
+Ben Swartzlander (NetApp)
+
+## Volume Populators
```
Comment (sftim) [inline content/en/blog/_posts/2022-04-12-volume-populators-beta.md:11]: Please omit this heading - we don't start articles with headings.

File: content/en/blog/_posts/2022-04-12-volume-populators-beta.md [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+
+```shell
+kubectl wait --for=condition=Complete job/example-job
+```
+
+And last examine the log from the job.
+
+```shell
+kubectl logs job/example-job
+Hello, world!
```
Comment (sftim) [inline content/en/blog/_posts/2022-04-12-volume-populators-beta.md:129]: It's best not to include the output in the box: `Hello, world!` isn't shell that people should run. Could you start a new block (see [Separate commands from output](https://kubernetes.io/docs/contribute/style/style-guide/#separate-commands-from-output) for an example)

**Review by sftim — COMMENTED** [review]
Some nits.

@katcosgrove I think the next steps are:
- tech review by SIG Storage
- assigning a publication date

**Comment by mickeyboxell** [comment]
@bswartz could we plan to have this ship on 16-MAY-2022?

**Comment by bswartz** [comment]
Sorry for extra commit, I keep rebasing on the wrong branch.

**Comment by sftim** [comment]
Article looks good. It'd be nice to have a SIG Storage person confirm that this is all technically correct.

LGTM
/approve

**Pull request merged**